### PR TITLE
PUBDEV-7980 GBM Interaction Constraints - R/Python API, doc

### PR DESCRIPTION
**Pull request by maurever**
JIRA: https://h2oai.atlassian.net/browse/PUBDEV-7980

### CODE REVIEW

**Comment by michalkurka** [comment]
@maurever is this PR ready for review? it looks good to me - however, it needs to be rebased it seems like there is a conflict with the parent PR

**Comment by maurever** [comment]
> @maurever is this PR ready for review? it looks good to me - however, it needs to be rebased it seems like there is a conflict with the parent PR

Yes, it is now ready for review. I was waiting on approval of the parent PR and then I rebased to the parent branch. So please, approve it if you think it is ok. :)

**Review by valenad1 — APPROVED** [review]
👍

**Pull request merged**
